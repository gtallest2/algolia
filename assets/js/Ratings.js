import React from 'react'

const { func, number } = React.PropTypes

const Ratings = React.createClass({
  propTypes: {
    currentRatingFilter: number,
    handleRatingFilter: func
  },
  handleRatingFilter (value) {
    this.props.handleRatingFilter(value)
  },
  render () {
    return (
      <div className='ratings-filter'>
        <h4>Rating</h4>
        <ul className='ratings'>
          <li onClick={this.handleRatingFilter.bind(null, 0)} className={this.props.currentRatingFilter === 0 ? 'active stars no-star' : 'stars no-star'} />
          <li onClick={this.handleRatingFilter.bind(null, 1)} className={this.props.currentRatingFilter === 1 ? 'active stars one-star' : 'stars one-star'} />
          <li onClick={this.handleRatingFilter.bind(null, 2)} className={this.props.currentRatingFilter === 2 ? 'active stars two-star' : 'stars two-star'} />
          <li onClick={this.handleRatingFilter.bind(null, 3)} className={this.props.currentRatingFilter === 3 ? 'active stars three-star' : 'stars three-star'} />
          <li onClick={this.handleRatingFilter.bind(null, 4)} className={this.props.currentRatingFilter === 4 ? 'active stars four-star' : 'stars four-star'} />
          <li onClick={this.handleRatingFilter.bind(null, 5)} className={this.props.currentRatingFilter === 5 ? 'active stars five-star' : 'stars five-star'} />
        </ul>
      </div>
    )
  }
})

export default Ratings
