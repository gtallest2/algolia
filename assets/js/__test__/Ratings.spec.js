import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Ratings from '../Ratings'

test('Ratings snapshot test', () => {
  const component = shallow(<Ratings />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('Ratings adds class of active to correct list element', () => {
  const component = shallow(
    <Ratings currentRatingFilter={3} />
  )
  expect(component.find('.active').exists()).toEqual(true)
  expect(component.find('.three-star').hasClass('active')).toEqual(true)
})

// test('Ratings sends correct value onClick', () => {
//   const component = shallow(
//     <Ratings handleRatingFilter={value => value } />
//   )
//   component.find('.two-star').simulate('click')
//   console.log(component);
//   // expect(component.find('.one-star').simulate('click')).toEqual(1)
// })
