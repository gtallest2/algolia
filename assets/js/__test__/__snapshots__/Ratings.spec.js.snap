exports[`test Ratings snapshot test 1`] = `
<div
  className="ratings-filter">
  <h4>
    Rating
  </h4>
  <ul
    className="ratings">
    <li
      className="stars no-star"
      onClick={[Function]} />
    <li
      className="stars one-star"
      onClick={[Function]} />
    <li
      className="stars two-star"
      onClick={[Function]} />
    <li
      className="stars three-star"
      onClick={[Function]} />
    <li
      className="stars four-star"
      onClick={[Function]} />
    <li
      className="stars five-star"
      onClick={[Function]} />
  </ul>
</div>
`;
